server/channels/api4/report.go

```diff
@@ -85,3 +88,46 @@ func getUsersForReporting(c *Context, w http.ResponseWriter, r *http.Request) {
 		c.Logger.Warn("Error writing response", mlog.Err(jsonErr))
 	}
 }
+
+func startUsersBatchExport(c *Context, w http.ResponseWriter, r *http.Request) {
+	if !(c.IsSystemAdmin() && c.App.SessionHasPermissionTo(*c.AppContext.Session(), model.PermissionSysconsoleReadUserManagementUsers)) {
+		c.SetPermissionError(model.PermissionSysconsoleReadUserManagementUsers)
+		return
+	}
+
+	if err := c.App.StartUsersBatchExport(c.AppContext, r.URL.Query().Get("date_range")); err != nil {
```

Comment:
So we can, but we would then need to convert it back to a string afterwards anyways to append it to the job data, so we would be undoing some work.

We could I suppose parse out the timestamps in the API layer, but I think the location of doing that is kind of arbitrary for this case, since the actual timestamps don't need to be validated in the API and they don't mean anything to any logic in the API at this point either.

We could also validate the string at the API layer, but that would also just error out in the job, and would only really only happen if someone called the API directly, and you would still get an error at that point.

Reply:
> So we can, but we would then need to convert it back to a string afterwards anyways to append it to the job data, so we would be undoing some work.

I actually think this would be a net win -- we wouldn't ideally be persisting raw data from the enduser as job metadata anyway. Besides any unexpected security concerns, this decouples the "schema" of this parameter between the two use cases: how we accept user input, and how we store offset metadata in the job database. One could imagine a future scenario where we want to programmatically generate reports, and be forced to stitch together a string to communicate this information instead of passing it explicitly.

Reply:
Okay, I would say it still seems a bit redundant, but from an architecture perspective it does make sense to always have some kind of "translation" happening between the API and the app logic, so the redundancy will apply in some cases.

I'll change it to compute the date range here.